verifier/src/lib.rs

```diff
@@ -10,67 +10,72 @@ pub use report_body::{
     MiscellaneousSelectVerifier, MrEnclaveVerifier, MrSignerVerifier, ReportDataVerifier,
 };
 
-use core::fmt::Debug;
+use core::fmt::{Debug, Display, Formatter};
 use mc_sgx_core_types::{
     Attributes, ConfigId, ConfigSvn, IsvSvn, MiscellaneousSelect, MrEnclave, MrSigner, ReportData,
 };
 use subtle::CtOption;
 
+/// An error that implements the [`Display`] trait.
+pub trait DisplayableError: Display + Clone {}
+
 /// Failed to verify.
-#[derive(Debug, Eq, PartialEq)]
+#[derive(displaydoc::Display, Debug, Eq, PartialEq, Clone)]
 pub enum VerificationError {
     /// A general error.
     General,
-    /// The attributes did not match expected:{expected} actual:{actual}
+    /// Forced failure via `AlwaysFalse`
+    AlwaysFalse,
```

Comment:
Added this as it felt odd to have "A general error" in the tree output instead of a more informative always false indicator, still probably a bit difficult for a user to understand though :(